src/expr.cpp

```diff
@@ -2084,14 +2090,15 @@ llvm::Value *BinaryExpr::GetValue(FunctionEmitContext *ctx) const {
         return nullptr;
     }
 
+    const bool isSigned = !arg0->GetType()->IsUnsignedType() || !arg1->GetType()->IsUnsignedType();
```

Comment:
`this->GetType()` should be used instead - it has all the logic for deducting the type of operation. And it's should not be duplicated here.